core/src/test/java/io/questdb/griffin/O3PartitionPurgeTest.java

```diff
@@ -0,0 +1,590 @@
+                int len = path.length();
+                for (int i = 0; i < 3; i++) {
+                    path.trimTo(len).put(".").put(Integer.toString(i)).concat("x.d").$();
+                    Assert.assertFalse(Chars.toString(path), Files.exists(path));
+                }
+            }
+        });
+    }
+
+    @Test
+    public void testAsyncPurgeOnBusyWriter() throws Exception {
+        int tableCount = 3;
+        assertMemoryLeak(() -> {
+            for (int i = 0; i < tableCount; i++) {
+                compiler.compile("create table tbl" + i + " as (select x, cast('1970-01-01T10' as timestamp) ts from long_sequence(1)) timestamp(ts) partition by DAY", sqlExecutionContext);
+            }
+
+            final CyclicBarrier barrier = new CyclicBarrier(3);
+            AtomicInteger done = new AtomicInteger();
+            // Open a reader so that writer will not delete partitions easily
+            ObjList<TableReader> readers = new ObjList<>(tableCount);
+            for (int i = 0; i < tableCount; i++) {
+                readers.add(engine.getReader(sqlExecutionContext.getCairoSecurityContext(), "tbl" + i));
+            }
+
+            Thread writeThread = new Thread(() -> {
+                try {
+                    barrier.await();
+                    for (int i = 0; i < 32; i++) {
+                        for (int j = 0; j < tableCount; j++) {
+                            compiler.compile("insert into tbl" + j +
+                                            " select 2, '1970-01-10T10' from long_sequence(1) " +
+                                            "union all " +
+                                            "select 1, '1970-01-09T09'  from long_sequence(1)"
+                                    , sqlExecutionContext);
+                        }
+                    }
+                    done.incrementAndGet();
+                    Path.clearThreadLocals();
+                } catch (Throwable ex) {
+                    LOG.error().$(ex).$();
+                    done.decrementAndGet();
+                }
+            });
+
+            Thread readThread = new Thread(() -> {
+                try {
+                    barrier.await();
+                    while (done.get() == 0) {
+                        for (int i = 0; i < tableCount; i++) {
+                            readers.get(i).openPartition(0);
+                            readers.get(i).reload();
+                        }
+                        LockSupport.parkNanos(0);
+                        Path.clearThreadLocals();
+                    }
+                } catch (Throwable ex) {
+                    LOG.error().$(ex).$();
+                    done.addAndGet(-2);
+                }
+            });
+
+            writeThread.start();
+            readThread.start();
+
+            barrier.await();
+            while (done.get() == 0) {
+                runPartitionPurgeJobs();
+                LockSupport.parkNanos(0);
```

Comment:
nit: `LockSupport.parkNanos(0)` is a no-op, so we can remove this line or increase the timeout: https://github.com/openjdk/jdk/blob/1725f77bcd6528d56960a0796fcea3725cc98b6a/src/java.base/share/classes/java/util/concurrent/locks/LockSupport.java#L375

Reply:
changed in https://github.com/questdb/questdb/pull/1778/commits/3b8609ba3442fa0a590cd19693726b231d02cbff